src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -50,18 +50,31 @@ type Client struct {
 	l net.Listener
 }
 
-// ListenAndServe makes the client listen for HTTP connections.  onListeningFn
-// is a callback that gets invoked as soon as the server is accepting TCP
-// connections.
+// Addr returns the address at which the client is listening, blocking until the
+// given timeout for an address to become available.
+func Addr(timeout time.Duration) (interface{}, bool) {
+	return addr.Get(timeout)
+}
+
+func (c *Client) Addr(timeout time.Duration) (interface{}, bool) {
+	return Addr(timeout)
+}
+
+// ListenAndServe makes the client listen for HTTP connections at a random port
+// on localhost. onListeningFn is a callback that gets invoked as soon as the
+// server is accepting TCP connections.
 func (client *Client) ListenAndServe(onListeningFn func()) error {
+	log.Debug("About to listen")
 	var err error
 	var l net.Listener
 
-	if l, err = net.Listen("tcp", client.Addr); err != nil {
-		return fmt.Errorf("Client proxy was unable to listen at %s: %q", client.Addr, err)
+	if l, err = net.Listen("tcp", "localhost:0"); err != nil {
```

Comment:
We could try 8787 and then use a random port if we can't bind... We should probably at least support running on 0.0.0.0 if the user specifies it from the command line though (even though I believe that's broken on the current valencia).